gravitee-am-identityprovider/gravitee-am-identityprovider-api/src/main/java/io/gravitee/am/identityprovider/api/oidc/OpenIDConnectIdentityProviderConfiguration.java

```diff
@@ -40,4 +41,13 @@ public interface OpenIDConnectIdentityProviderConfiguration extends SocialIdenti
     default String getClientAuthenticationMethod() {
         return ClientAuthenticationMethod.CLIENT_SECRET_POST;
     }
+
+    /**
+     * Type of challenge used for PKCE. null means PKCE shouldn't be used
+     */
+    CodeChallengeMethod getCodeChallengeMethod();
```

Comment:
**todo**: can you please provide default implementation which return null to avoid having to release all EE plugins which are inherting from this config interface, it will also help to avoid error when a customer will update the service but not the plugin ?